src/main/java/securbank/controller/AdminController.java

```diff
@@ -111,9 +108,69 @@ public String getUserDetail(Model model, @PathVariable UUID id) {
 
 		model.addAttribute("user", user);
 		logger.info("GET request: Internal user details by id");
-
-		return "admin/userdetail";
-	}
+        	
+        return "admin/userdetail";
+    }
+	
+	@GetMapping("/admin/user/request")
+    public String getAllUserRequest(Model model) {
+		List<ModificationRequest> modificationRequests = userService.getModificationRequests("pending", "internal");
+		if (modificationRequests == null) {
+			model.addAttribute("modificationrequests", new ArrayList<ModificationRequest>());
+		}
+		else {
+			model.addAttribute("modificationrequests", modificationRequests);	
+		}
+		logger.info("GET request: All user requests");
+		
+        return "admin/modificationrequests";
+    }
+	
+	@GetMapping("/admin/user/request/{id}")
+    public String getUserRequest(Model model, @PathVariable() UUID id) {
+		ModificationRequest modificationRequest = userService.getModificationRequest(id);
+		
+		if (modificationRequest == null) {
+			return "redirect:/error?=code=400&path=request-invalid";
+		}
+		model.addAttribute("modificationrequest", modificationRequest);
+		logger.info("GET request: User modification request by ID");
+		
+		
+        return "admin/modificationrequest_detail";
+    }
+	
+	@PostMapping("/admin/user/request/{requestId}")
+    public String approveEdit(@PathVariable UUID requestId, @ModelAttribute ModificationRequest request, BindingResult bindingResult) {
+		String status = request.getStatus();
+		if (status == null || !(request.getStatus().equals("approved") || !request.getStatus().equals("rejected"))) {
```

Comment:
When you reject a request it gives an error. The logic isn't right here. The ! outside is fine but there shouldn't be another one inside for 'rejected'.